mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Utils.kt

```diff
@@ -0,0 +1,24 @@
+package ch.epfl.sdp.mobile.test.state
+
+import android.app.Activity
+import android.app.Instrumentation.ActivityResult
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.matcher.IntentMatchers
+
+/**
+ * Executes the given [block] by returning an [ActivityResult] with the code
+ * [Activity.RESULT_CANCELED] each time an intent is triggered. This simulates devices which do not
+ * support AR.
+ *
+ * @param block the block of code to execute.
+ */
+fun withCanceledIntents(block: () -> Unit) {
```

Comment:
Just FYI, this would be an excellent candidate for an `inline` function. If we ever want to use `withCanceledIntents` with some suspending content, making the function `inline` means we can have a single function for both suspending and non-suspending content.

Also, since this is some test code, making the function `inline` would not decrease Jacoco coverage.

```suggestion
inline fun withCanceledIntents(block: () -> Unit) {
```

Reply:
Fix in commit https://github.com/epfl-SDP/android/pull/251/commits/20d13676969458200a45a10a7be177de640dfd83